File: java/src/main/java/com/example/demo/api/tb/domain/TaskType.java
package com.example.demo.api.tb.domain;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * 任务类型实体类
 * @TableName task_type
 */
@TableName(value = "task_type")
@Data
public class TaskType implements Serializable {

    /**
     * 主键ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 类型名称
     */
    @TableField // value = "name" 可省略，默认匹配字段名
    private String name;

    /**
     * 描述信息
     */
    @TableField // value = "description"
    private String description;

    /**
     * 所属项目ID
     */
    @TableField(value = "project_id")
    private Long projectId;

    /**
     * 是否隐藏：0-隐藏，1-true-显示
     */
    @TableField(value = "is_hidden")
    private Boolean isHidden;


    /**
     * 创建人ID
     */
    @TableField(value = "create_by", fill = FieldFill.INSERT)
    private Long createBy;

    /**
     * 创建时间
     */
    @TableField(value = "create_time", fill = FieldFill.INSERT)
    private Date createTime;

    /**
     * 更新时间
     */
    @TableField(value = "update_time", fill = FieldFill.INSERT_UPDATE)
    private Date updateTime;

    /**
     * 逻辑删除 默认是 0
     */
    @TableField(value = "is_deleted")
    @TableLogic
    private Integer isDeleted;

    /**
     * 序列化版本标识
     */
    @TableField(exist = false)
    private static final long serialVersionUID = 1L;
}

File: fronted/src/types/task-type.ts
// 任务类型相关类型定义

// 任务类型接口
export interface TaskType {
  id: number
  name: string
  description: string
  projectId: number
  createBy: number
  createTime: string
  updateTime: string
  status: number // 1: 启用, 0: 禁用
  // 兼容后端返回的隐藏字段
  isHidden?: boolean
}

// 创建任务类型请求参数
export interface CreateTaskTypeRequest {
  name: string
  description?: string
  projectId: number
}

// 保存任务类型模板请求参数（后端 /taskTemplate/saveTaskType）
export interface SaveTaskTypeRequest {
  name: string
  description?: string
  projectId: number
  isHidden: boolean
}

// 更新任务类型请求参数
export interface UpdateTaskTypeRequest {
  name?: string
  description?: string
  // 兼容模板更新接口
  isHidden?: boolean
}

// 任务类型列表响应
export interface TaskTypeListResponse {
  code: number
  msg: string
  data: TaskType[]
}

// 任务类型详情响应
export interface TaskTypeDetailResponse {
  code: number
  msg: string
  data: TaskType
}

// 删除任务类型响应
export interface DeleteTaskTypeResponse {
  code: number
  msg: string
  data: null
}